src/graph/transform/partition_hetero.cc

```diff
@@ -0,0 +1,259 @@
+        std::vector<int64_t> vec;
+        vec.push_back(i);
+        part_map[part_id] = vec;
+      } else {
+        it->second.push_back(i);
+      }
+    }
+    std::vector<int> part_ids;
+    std::vector<std::vector<int64_t>> part_nodes;
+    int max_part_id = 0;
+    for (auto it = part_map.begin(); it != part_map.end(); it++) {
+      max_part_id = std::max(it->first, max_part_id);
+      part_ids.push_back(it->first);
+      part_nodes.push_back(it->second);
+    }
+    // When we construct subgraphs, we only access in-edges.
+    // We need to make sure the in-CSR exists. Otherwise, we'll
+    // try to construct in-CSR in openmp for loop, which will lead
+    // to some unexpected results.
+    ugptr->GetInCSR();
+    std::vector<std::shared_ptr<HaloHeteroSubgraph>> subgs(max_part_id + 1);
+    int num_partitions = part_nodes.size();
+#pragma omp parallel for
+    for (int i = 0; i < num_partitions; i++) {
+      auto nodes = aten::VecToIdArray(part_nodes[i]);
+      HaloHeteroSubgraph subg = GetSubgraphWithHalo(hgptr, nodes, num_hops);
+      std::shared_ptr<HaloHeteroSubgraph> subg_ptr(
+        new HaloHeteroSubgraph(subg));
+      int part_id = part_ids[i];
+      subgs[part_id] = subg_ptr;
+    }
+    List<HeteroSubgraphRef> ret_list;
+    for (size_t i = 0; i < subgs.size(); i++) {
+      ret_list.push_back(HeteroSubgraphRef(subgs[i]));
+    }
+    *rv = ret_list;
+  });
+
+// TODO(JJ): What's this?
+DGL_REGISTER_GLOBAL("partition._CAPI_DGLReassignEdges_Hetero")
+  .set_body([](DGLArgs args, DGLRetValue *rv) {
+    HeteroGraphRef g = args[0];
+    auto hgptr = std::dynamic_pointer_cast<HeteroGraph>(g.sptr());
+    CHECK(hgptr) << "Invalid HeteroGraph object";
+    CHECK_EQ(hgptr->relation_graphs().size(), 1)
+      << "Reorder only supports HomoGraph";
+    auto ugptr = hgptr->relation_graphs()[0];
+    auto csrmat = ugptr->GetCSRMatrix(0);
+    int64_t num_edges = csrmat.data->shape[0];
+    IdArray new_data =
+      IdArray::Empty({num_edges}, csrmat.data->dtype, csrmat.data->ctx);
+    // Return the original edge Ids.
+    *rv = new_data;
+    // TODO(zhengda) I need to invalidate out-CSR and COO.
+
+    // Generate new edge Ids.
+    // TODO(zhengda) after assignment, we actually don't need to store them
+    // physically.
+    ATEN_ID_TYPE_SWITCH(new_data->dtype, IdType, {
+      IdType *typed_new_data = static_cast<IdType *>(new_data->data);
+      IdType *typed_data = static_cast<IdType *>(csrmat.data->data);
+      for (int64_t i = 0; i < num_edges; i++) {
+        typed_new_data[i] = typed_data[i];
+        typed_data[i] = i;
+      }
+    });
+  });
+
+DGL_REGISTER_GLOBAL("partition._CAPI_GetHaloSubgraphInnerNodes_Hetero")
+  .set_body([](DGLArgs args, DGLRetValue *rv) {
+    HeteroSubgraphRef g = args[0];
+    auto gptr = std::dynamic_pointer_cast<HaloHeteroSubgraph>(g.sptr());
+    CHECK(gptr) << "The input graph has to be HaloHeteroSubgraph";
+    *rv = gptr->inner_nodes[0];
+  });
```

Comment:
most of the code here is copied from the old code?

Reply:
Yes